bin/sozo/src/commands/options/transaction.rs

```diff
@@ -58,22 +80,186 @@ impl TransactionOptions {
             (false, false) => Ok(TxnAction::Send {
                 wait: self.wait || self.walnut,
                 receipt: self.receipt,
-                max_fee_raw: self.max_fee_raw,
-                fee_estimate_multiplier: self.fee_estimate_multiplier,
+                fee_config: match self.fee_token {
+                    FeeToken::Strk => {
+                        FeeConfig::Strk(StrkFeeConfig { gas: self.gas, gas_price: self.gas_price })
+                    }
+                    FeeToken::Eth => FeeConfig::Eth(EthFeeConfig {
+                        max_fee_raw: self.max_fee_raw,
+                        fee_estimate_multiplier: self.fee_estimate_multiplier,
+                    }),
+                },
                 walnut: self.walnut,
             }),
         }
     }
 }
 
-impl From<TransactionOptions> for TxnConfig {
-    fn from(value: TransactionOptions) -> Self {
-        Self {
-            fee_estimate_multiplier: value.fee_estimate_multiplier,
+impl TryFrom<TransactionOptions> for TxnConfig {
+    type Error = anyhow::Error;
+
+    fn try_from(value: TransactionOptions) -> Result<Self> {
+        match value.fee_token {
+            FeeToken::Eth => {
+                if value.gas.is_some() || value.gas_price.is_some() {
+                    bail!(
+                        "Gas and gas price are not supported for ETH transactions. Use \
+                         `--fee-token strk` instead or use `--max-fee-raw` and \
+                         `--fee-estimate-multiplier`."
+                    );
+                }
+            }
+            FeeToken::Strk => {
+                if value.max_fee_raw.is_some() || value.fee_estimate_multiplier.is_some() {
+                    bail!(
+                        "Max fee raw and fee estimate multiplier are not supported for STRK \
+                         transactions. Use `--fee-token eth` instead or use `--gas` and \
+                         `--gas-price`."
+                    );
+                }
+            }
+        };
```

Comment:
@glihm nvm i was wrong, it seems to be required because there seems to be now way to prevent using strk specific args if we set `--fee-token eth` or vice versa.